src/synthesiser/Synthesiser.cpp

```diff
@@ -197,6 +197,70 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             };
         }
 
+        std::pair<std::stringstream, std::stringstream> getPaddedRangeBounds(const RamRelation& rel,
+                const std::vector<RamExpression*>& rangePatternLower,
+                const std::vector<RamExpression*>& rangePatternUpper) {
+            std::stringstream low;
+            std::stringstream high;
+
+            size_t realArity = rel.getArity();  // making this distinction for provenance
+            size_t arity = rangePatternLower.size();
+
+            low << "Tuple<RamDomain," << realArity << ">{{";
+            high << "Tuple<RamDomain," << realArity << ">{{";
+
+            for (size_t column = 0; column < arity; column++) {
+                std::string supremum;
+                std::string infimum;
+
+                switch (rel.getAttributeTypes()[column][0]) {
+                    /*
+            case 'i':
```

Comment:
Don't leave in commented out code. A todo can be left here instead of the switch.